tests/Unit/Domain/Services/TalkRating/YesNoRatingTest.php

```diff
@@ -27,28 +27,61 @@ class YesNoRatingTest extends \PHPUnit\Framework\TestCase
     use MockeryPHPUnitIntegration;
 
     /**
-     * @dataProvider ratingProvider
+     * @dataProvider validRatingProvider
      */
-    public function testValidRatings($rating, $valid)
+    public function testValidRatings($rating)
+    {
+        $mockAuth = Mockery::mock(Authentication::class)->shouldIgnoreMissing();
+        $metaMock = Mockery::mock(TalkMeta::class);
+
+        $yesno    = new YesNoRating($metaMock, $mockAuth);
+
+        $this->assertTrue($yesno->isValidRating($rating));
+    }
+
+    /**
+     * @dataProvider invalidRatingProvider
+     */
+    public function testInvalidRatings($rating)
+    {
+        $mockAuth = Mockery::mock(Authentication::class)->shouldIgnoreMissing();
+        $metaMock = Mockery::mock(TalkMeta::class);
+
+        $yesno    = new YesNoRating($metaMock, $mockAuth);
+
+        $this->assertFalse($yesno->isValidRating($rating));
+    }
+
+    public function testGetRatingNameReturnsYesNo()
     {
         $mockAuth = Mockery::mock(Authentication::class);
         $mockAuth->shouldReceive('userId');
         $metaMock = Mockery::mock(TalkMeta::class);
+
         $yesno    = new YesNoRating($metaMock, $mockAuth);
-        $this->assertSame($valid, $yesno->isValidRating($rating));
+
+        $this->assertSame('YesNo', $yesno->getRatingName());
+    }
+
+    public function validRatingProvider(): array
+    {
+        return [
+            [-1],
```

Comment:
Not sure, do `-1` and `-0` even make sense?

For a `YesNoRating` strategy I would expect to accept only two values, either `0` or `1`, or `false` and `true`.

What do you think?

Reply:
How this has worked before was: -1 = dislike, 1 = like, 0 = no opinion/ not rated. Changing this around would cause BC to break

Reply:
Thank you for clarifying, @BackEndTea!